app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -11,12 +12,20 @@ def cloud_instance
     @cloud_instance ||= cloud_instances_api.pcloud_cloudinstances_get(cloud_instance_id)
   end
 
-  def pvm_instances
-    @pvm_instances ||= pvm_instances_api.pcloud_pvminstances_getall(cloud_instance_id).pvm_instances || []
+  def pvm_instance(pvm_instance_id)
+    pvm_instances_by_id[pvm_instance_id] ||= pvm_instances_api.pcloud_pvminstances_get(cloud_instance_id, pvm_instance_id)
+  rescue IbmCloudPower::ApiError => err
+    error_message = JSON.parse(err.response_body)["description"]
+    _log.debug("PVMInstanceID not found: #{error_message}")
```

Comment:
These debug logs were helpful as I was working on this, but I don't see too many others around. Is the parsing/_log.debug call expensive?